controllers/discovery/searchTopic.js

```diff
@@ -42,6 +50,7 @@ const Search = async (req, res) => {
         type: QueryTypes.SELECT,
         replacements: {
           userId,
+          anonymousUserId: anonymousUser?.user_id,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `anonymousUserId` is passed correctly to the replacements object. However, if the `findAnonymousUserId` function fails or returns null, this could potentially cause issues. It would be better to add a check before executing the query.
```diff
+           anonymousUserId: anonymousUser?.user_id,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->